### ISSUE TITLE
In iOS 14 Updated devices injectedJavascript callBack not receiving Webview onMessage Method

### ISSUE THREAD

**Issue by Sinson48** [react-native-webview/react-native-webview#1812]
**Bug description:**
 In iOS 14 Updated devices InjectedJavascript callback not receiving Webview onMessage Method.
The other iOS lower version (< 14 )of devices it working as expected and the android device also it working five.

**Expected behavior:**
Need to work on all latest iOS version updated devices.
--->

**My Code:**

const INJECTEDJAVASCRIPT = `
      const meta = document.createElement('meta');
      window.checkoutError = function(){ window.ReactNativeWebView.postMessage('checkout-error') };
      window.checkoutSuccess = function(){ window.ReactNativeWebView.postMessage('checkout-success') };
      const script = document.createElement("script");
      script.setAttribute("data-error", "checkoutError");
      script.setAttribute("data-success", "checkoutSuccess");
      script.onload = function() { checkOut() };
      document.getElementsByTagName('head')[0].appendChild(script);
    `;
    
    <WebView
          source={{ html }}
          originWhitelist={['*']}
          javaScriptEnabled={true}
          onNavigationStateChange={this._onNavigationStateChange}
          onMessage={this.handleMessage}
          injectedJavaScript={INJECTEDJAVASCRIPT}
          mixedContentMode={'always'}
        />

 handleMessage = (d) => {
    const { navigation } = this.props;
    const msg = d.nativeEvent.data;
  switch (msg) {
      case 'checkout-success':
        return his.paymentSuccess();
      case 'checkout-error':
        return navigation.goBack();
      default:
        return console.log(`ONMESSAGE: `, msg);
    }
    }

**Environment:**
 - OS: iOS
 - OS version: iOS 14
 - react-native version: 0.63.3
 - react-native-webview version: 11.0.2

**Comment by kyungjoongo** [comment]
postMessage doesn't seem to work in ios14 version.
It works fine on Android.

**Comment by danielfx90** [comment]
Hi, I tried it out on iOS 14 and it works fine.
```
const INJECTED_JAVASCRIPT = `(function() {
  window.ReactNativeWebView.postMessage(JSON.stringify(window.location));
})();`;
```
```
<WebView
  source={{ uri: 'https://google.com' }}
  injectedJavaScript={INJECTED_JAVASCRIPT}
  onMessage={(msg) => console.log(msg)}
/>
```

**Comment by toastman** [comment]
I have same issue ( 
and @danielfx90 fix didn't work for me

**Comment by parkcoop** [comment]
Also having this issue.

**Comment by marksyzm** [comment]
Appears to be a react-dom issue? Observer is undefined. 
> TypeError: undefined is not an object (evaluating 'window.webkit.messageHandlers.observer.postMessage')
<img width="996" alt="Screenshot 2021-03-03 at 15 09 35" src="https://user-images.githubusercontent.com/420864/109826232-7e0e2f80-7c32-11eb-873e-cc4f9cd11118.png">

**Comment by marksyzm** [comment]
^^ That is the output from the WebView component, not the main JSContext

**Comment by marksyzm** [comment]
Tested in 14.4 and 14.3: not working. Going to test 13.* now.

**Comment by toastman** [comment]
I solve my problem via `onNavigationStateChange` - track changes in webView url and based on needed condition fire related  action.

**Comment by marksyzm** [comment]
Wow, that’s a horrible workaround but I’ll take it

**Comment by marksyzm** [comment]
@danielfx90 the problem I am noticing is that as soon as you put the post message reference in a function, it then breaks, but your example where it is exposed doesn’t. I’ll stick together an example of mine tomorrow.

**Comment by danielfx90** [comment]
> @danielfx90 the problem I am noticing is that as soon as you put the post message reference in a function, it then breaks, but your example where it is exposed doesn’t. I’ll stick together an example of mine tomorrow.

Any reason why mine doesn't break? I haven't touched the library since then, and therefore did not look further into this issue, but the feature on our app is still working. It is certainly odd.

**Comment by marksyzm** [comment]
Here's what breaks for me:

> TypeError: undefined is not an object (evaluating 'window.webkit.messageHandlers.observer.postMessage')

```jsx
const CLOSE_MESSAGE = 'close';
const onMessage = useCallback(
    (event) => {
      const {
        nativeEvent: { data },
      } = event;

      if (data === CLOSE_MESSAGE) {
        navigation.goBack();
      }
    },
    [navigation],
  );

  const INJECTED_JAVASCRIPT = `
  (function() {
    // This works fine, as does running the function in the safari console
    window.ReactNativeWebView.postMessage("${CLOSE_MESSAGE}");
    // This doesn't - it's set in the window of the webview but firing it causes the error mentioned.
    // Running it directly in the browser does work.
    window.onClose = () => {
      window.ReactNativeWebView.postMessage("${CLOSE_MESSAGE}");
    };
  })();
  `;

  return (
    <WrappedWebView
      source={{ uri: `https://someurl.com/` }}
      injectedJavaScript={INJECTED_JAVASCRIPT}
      onMessage={onMessage}
    />
  );
  ```

**Comment by marksyzm** [comment]
Nope, it was an issue with something else entirely on the front end. Ignore all the above from me.

**Comment by marksyzm** [comment]
I'll just leave a note that this may be the cause though:

```
window.checkoutSuccess = function(){
  window.ReactNativeWebView.postMessage('checkout-success');
};
```
As opposed to this:
```
window.ReactNativeWebView.postMessage('checkout-success');
```
I've a feeling the first one is silently failing.
